src/RModularityOptimizer.cpp

```diff
@@ -82,8 +82,8 @@ IntegerVector RunModularityClusteringCpp(Eigen::SparseMatrix<double> SNN,
     if(node1.size() == 0) {
       stop("Matrix contained no network data.  Check format.");
     }
-
-    network = matrixToNetwork(node1, node2, edgeweights, modularityFunction);
+    int nNodes = std::max(SNN.cols(), SNN.rows());
```

Comment:
As I understand it, SNN is the weighted adjacency matrix, so it ought to be square. Maybe you should require that it is square to prevent further issues down the line?

Reply:
It's a good point but I think checking for that should probably happen upstream of this function. Originally back in the v1 days, for this clustering functionality we were calling out to a java implementation provided by the authors of the clustering papers (http://www.ludowaltman.nl/slm/). It was then basically directly translated to this c++ implementation which was nice for both speed and it made distribution/installation easier. While we almost always tend to use it with the SNN matrix, the clustering algorithms themselves don't require the input network to have that structure and some of the tests that we have for it (in https://github.com/satijalab/seurat/blob/master/tests/testthat/test_modularity_optimizer.R)  don't actually use a square matrix as input.